src/ui/AriaLive/AriaLive.ts

```diff
@@ -3,13 +3,16 @@ import { $$ } from '../../utils/Dom';
 import { QueryEvents, IQuerySuccessEventArgs, IQueryErrorEventArgs } from '../../events/QueryEvents';
```

Comment:
General question:

Is aria-live the correct approach?

My understanding is aria-live is for announcing updates coming from a part of the app different from where the user is focused. e.g.

- Executing a search updates results/query summary -> result list is different from search box.
- Selecting a facet value updates result/query summary -> result list is different from facet.
- Push notifications -> user is focused elsewhere when one arrives.

In the case of query suggestions though, suggestions would only update if a user is focused on the search box -> the same part of the interface?

So I wonder if it's possible to link the two elements? Perhaps [these examples](https://alphagov.github.io/accessible-autocomplete/examples/) could serve as a guide. I see they are using [aria-owns](https://www.digitala11y.com/aria-owns-properties/), and when I type, I hear and updated number of suggestions.

Reply:
Don't think you have control on what is said, thought. For me voice over just tells me X results are available. I wouldn't know if it's query suggestions or search results. I think from an impaired end user perspective, it's better to use aria-live

Reply:
It looks like in the example you gave, they are also using `aria-live`. I can find it in the page and in their source code, however, for some reason it always seems to be empty?

Reply:
Indeed 👀 I actually see two `aria-live`s (`autocomplete-default__status--A` and `autocomplete-default__status--B`) related to the first input, and the text alternates between them on every keystroke.

Not sure why they're doing that, but it runs counter to how I thought aria-live should be used. Going to approve 🤷

Reply:
In this situation, I do see `aria-live` as a workaround, not so much the best practice. However, it seems like only TalkBack gives an acceptable amount of information without `aria-live`.